lib/smb2/file.rb

```diff
@@ -84,34 +105,59 @@ def read(offset: 0, length: nil)
   # @param length [Fixnum] number of bytes to read, starting from `offset`. If
   #   this is greater than the server's maximum read size, the server will
   #   respond with STATUS_INVALID_PARAMETER
-  # @return [String]
-  def read_chunk(offset: 0, length: self.tree.client.max_read_size)
-    packet = Smb2::Packet::ReadRequest.new do |request|
-      request.read_offset = offset
-      request.read_length = length
-      request.file_id = self.create_response.file_id
-      request.minimum_count = 0
-    end
+  # @return [Smb2::Packet::ReadResponse]
+  def read_chunk(offset: self.pos, length: self.tree.client.max_read_size)
+    packet = Smb2::Packet::ReadRequest.new(
+      read_offset: offset,
+      read_length: length,
+      file_id: self.create_response.file_id,
+      minimum_count: 0
+    )
 
     response = tree.send_recv(packet)
 
     response_packet = Smb2::Packet::ReadResponse.new(response)
     @last_read_response = response_packet
 
-    response_packet.data
+    seek(response_packet.data_length, IO::SEEK_CUR)
+
+    response_packet
+  end
+
+  # Seeks to a given `offset` in the stream according to the value of
+  # `whence`:
+  #
+  # |`whence`||
+  # |----------------------|--------------------------------------------------
+  # | :CUR or IO::SEEK_CUR | Seeks to `amount` plus current position
+  # | :END or IO::SEEK_END | Seeks to `amount` plus end of stream (you probably want a negative value for `amount`)
+  # | :SET or IO::SEEK_SET | Seeks to the absolute location given by `amount`
+  #
+  # @see http://ruby-doc.org/core-2.2.2/IO.html#method-i-seek
+  # @param amount [Fixnum]
+  # @param whence [Symbol]
+  def seek(amount, whence=IO::SEEK_SET)
```

Comment:
You should add a `@return` tag.  Either `@return [Integer]` if returning `@pos` is part of the API or `@return [void]` if callers should not depend on the return value.  Right YARD documents it as the default `@return [Object]`:
![screen shot 2015-05-06 at 1 15 45 pm](https://cloud.githubusercontent.com/assets/2230482/7500214/0abf6b26-f3f2-11e4-881e-6fee7875d8bf.png)